x/evm/keeper/msg_server.go

```diff
@@ -84,3 +84,22 @@ func (k *Keeper) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// UpdateImplmentationProposal update proxy with new implementation for counsil addresses
+func (k *Keeper) UpdateImplmentationProposal(goCtx context.Context, msg *types.MsgUpdateImplmentationProposal) (*types.MsgUpdateImplmentationProposalResponse, error) {
+	if !k.accountKeeper.GetModuleAddress(govtypes.ModuleName).Equals(msg.GetSigners()[0]) {
```

Comment:
I have here concerns as signers were just taken from authority and I do not see any way how to protect with the new proposal approach who could execute this transaction (I assume that msg_server handler could be executed by anybody) Please correct me if I am wrong @alexstratos @Xiong-stratos